apps/taxes/src/modules/taxes/tax-provider-utils.ts

```diff
@@ -0,0 +1,16 @@
+/*
+ * The providers sdk types claim to sometimes return undefined.
+ * If it ever happens, we have nothing to fall back to, so we throw an error.
+ * Should only be used for values that are required for further calculation.
+ */
+function resolveOptionalOrThrow<T>(value: T | undefined): T {
+  if (value === undefined) {
+    throw new Error("Could not resolve data. Value needed for further calculation is undefined.");
+  }
+
+  return value;
+}
+
```

Comment:
@peelar in the future lets allow passing Error here, because this error will not be valuable when you check why webhook failed - we don't know what field was missing.

so you can have something like

```
taxValue: resolveOptionalOrThrow(resp.taxValue, new Error('Missing taxValue field from avalara'))
```

Reply:
I initially implemented it, but I ended up changing the logs so they display the stack trace once there is an error. I will add it